app/views/app_groups/show.html.slim

```diff
@@ -44,20 +47,25 @@ h4.mb-3 All Applications
     tbody
       - @apps.each do |barito_app|
         tr
-          td.text-center = barito_app.name
-          td.text-center = barito_app.topic_name
-          td.text-center = barito_app.secret_key
-          td.text-center = barito_app.max_tps
-          td.text-center = barito_app.status
-          td.text-center = barito_app.log_count
-          td.text-center = barito_app.created_at
-          td.text-center = link_to "Delete", app_path(barito_app), method: :delete, data: { confirm: 'Are you sure?' }, class: 'btn btn-danger'
-      tr
-        td.text-center = f.text_field :name
-        td.text-center = f.text_field :topic_name
-        td.text-center
-        td.text-center = f.text_field :max_tps
-        td.text-center
-        td.text-center
-        td.text-center
-        td.text-center = f.submit 'Create', class: 'btn btn-primary'
+          td.text-center= barito_app.name
+          td.text-center= barito_app.topic_name
+          td.text-center= barito_app.secret_key
+          td.text-center= barito_app.max_tps
+          td.text-center= barito_app.status
+          td.text-center= barito_app.log_count
+          td.text-center= barito_app.created_at
+          td.text-center
+            - if @allow_upgrade
```

Comment:
I don't understand, why we check for `allow_upgrade` before deleting?

Reply:
So I think the `delete` action is only allowed for `admin` and `owner` and then I just reused the variable to add authorization for `delete` action. Maybe I should create a new instance variable to avoid ambiguity.